cmake/linker_script/common/common-rom.cmake

```diff
@@ -7,6 +7,7 @@ zephyr_linker_section_obj_level(SECTION init LEVEL PRE_KERNEL_2)
 zephyr_linker_section_obj_level(SECTION init LEVEL POST_KERNEL)
 zephyr_linker_section_obj_level(SECTION init LEVEL APPLICATION)
 zephyr_linker_section_obj_level(SECTION init LEVEL SMP)
+zephyr_linker_section_obj_level(SECTION init LEVEL CUSTOM_BOOT_LEVEL)
```

Comment:
Can you explain why we need yet another init level? How is this related with this proposal?

Reply:
updated PR to get child device information via DT_FOREACH_CHILD_STATUS_OKAY_SEP and DEVICE_INIT_DT_GET macros. This will avoid walking through the section for getting right device object (using Hardware id etc.). So this is no more required and is removed.